File: app/src/main/java/cn/com/woong/readhub/ui/mine/set/SettingActivity.java
package cn.com.woong.readhub.ui.mine.set;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import com.blankj.utilcode.util.BarUtils;
import com.blankj.utilcode.util.SPUtils;

import butterknife.BindView;
import butterknife.ButterKnife;
import cn.com.woong.readhub.R;
import cn.com.woong.readhub.base.BaseActivity;
import cn.com.woong.readhub.constant.Constant;
import cn.com.woong.readhub.ui.widget.ArrowItemView;
import cn.com.woong.readhub.ui.widget.TitleBarLayout;

/**
 * @author wong
 *         Created by wong on 2018/5/7.
 */

public class SettingActivity extends BaseActivity {
    @BindView(R.id.title_bar)
    TitleBarLayout titleBar;
    @BindView(R.id.arrow_english)
    ArrowItemView arrowEnglish;

    public static void startSettingActivity(Activity activity) {
        Intent intent = new Intent(activity, SettingActivity.class);
        activity.startActivity(intent);
    }

    @Override
    protected int getLayoutId() {
        return R.layout.activity_setting;
    }

    @Override
    protected void initView() {
        BarUtils.setStatusBarColor(this, getResources().getColor(R.color.c1), 1);
        titleBar.setTitleBarBgColor(getResources().getColor(R.color.c1));
        titleBar.setTitleColor(getResources().getColor(R.color.b7));

        titleBar.setTitle(getString(R.string.setting));
        titleBar.setLeftBack(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });

        arrowEnglish.setSwitchButtonVisible(true);
    }

    @Override
    protected void initData() {
        arrowEnglish.setOnArrowListener(new ArrowItemView.OnArrowListener() {
            @Override
            public void onSwitchChange(boolean checkable) {
                SPUtils.getInstance().put(Constant.SP_SHOW_ENGLISH, checkable);
            }
        });
    }
}

File: app/src/main/java/cn/com/woong/readhub/ui/mine/SettingActivity.kt
package cn.com.woong.readhub.ui.mine

import com.blankj.utilcode.util.SPUtils
import cn.com.woong.readhub.ui.widget.ArrowItemView
import com.blankj.utilcode.util.BarUtils
import android.content.Intent
import android.app.Activity
import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import cn.com.woong.readhub.R
import cn.com.woong.readhub.constant.Constant
import kotlinx.android.synthetic.main.activity_setting.*

class SettingActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_setting)
        initView()
    }

    private fun initView() {
        BarUtils.setStatusBarColor(this, resources.getColor(R.color.c1), 1)
        title_bar.setTitleBarBgColor(resources.getColor(R.color.c1))
        title_bar.setTitleColor(resources.getColor(R.color.b7))

        title_bar.setTitle(getString(R.string.setting))
        title_bar.setOnClickListener { finish() }

        arrow_english.setSwitchButtonVisible(true)
        arrow_english.setOnArrowListener(object : ArrowItemView.OnArrowListener {
            override fun onSwitchChange(checkable: Boolean) {
                SPUtils.getInstance().put(Constant.SP_SHOW_ENGLISH, checkable)
            }
        })
    }

    companion object {
        fun startSettingActivity(context: Context) {
            val intent = Intent(context, SettingActivity::class.java)
            context.startActivity(intent)
        }
    }
}
